Add filterwarnings rules to the pytest section of pyproject.toml to ignore warnings from changes to
core_schemas types in https://github.com/pydantic/pydantic-core/pull/980

This can probably be removed once fastapi requires pydantic>2.4.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -108,6 +108,9 @@ filterwarnings = [
     "ignore::trio.TrioDeprecationWarning",
     # TODO remove pytest-cov
     'ignore::pytest.PytestDeprecationWarning:pytest_cov',
+    # CHANGE IN PYDANTIC, ignore warnings about pydantic-core validation methods
+    'ignore:`general_(before|after|wrap|plain)_validator_function` is deprecated.*:DeprecationWarning:',
+    'ignore:`FieldValidationInfo` is deprecated.*:DeprecationWarning:',
 ]

 [tool.coverage.run]
